src/Rings.js

```diff
@@ -0,0 +1,602 @@
+
+              if (edge[node].id === p.id) {
+                edge[node].edges[oppId] = {
+                  angle: p.radians + Math.PI,
+                  radius: ringWidth / 2
+                };
+              }
+              else {
+                edge[node].edges[oppId] = {
+                  angle: target.radians,
+                  radius: ringWidth / 2
+                };
+              }
+            });
+
+            edges.push(edge);
+          }
+        }
+      });
+    });
+
+    nodes.forEach(node => {
+
+      if (node.id !== this._center) {
+        const fontSize = this._shapeConfig.labelConfig.fontSize && this._shapeConfig.labelConfig.fontSize(node) || 11;
+        const lineHeight = fontSize * 1.4;
+        const height = lineHeight * 2;
+        const padding = 5;
+        const width = ringWidth - node.r;
+
+        let angle = node.radians * (180 / Math.PI);
+        let x = node.r + padding;
+        let textAnchor = "start";
+
+        if (angle < -90 || angle > 90) {
+          x = -node.r - width - padding;
+          textAnchor = "end";
+          angle += 180;
+        }
+
+        node.labelBounds = {
+          x,
+          y: -lineHeight / 2,
+          width,
+          height
+        };
+
+        node.rotate = angle;
+        node.textAnchor = textAnchor;
+      }
+      else {
+        node.labelBounds = {
+          x: -primaryRing / 2,
+          y: -primaryRing / 2,
+          width: primaryRing,
+          height: primaryRing
+        };
+      }
+    });
+
+    this._linkLookup = links.reduce((obj, d) => {
+      if (!obj[d.source.id]) obj[d.source.id] = [];
+      obj[d.source.id].push(d.target);
+      if (!obj[d.target.id]) obj[d.target.id] = [];
+      obj[d.target.id].push(d.source);
+      return obj;
+    }, {});
+
+    this._container = this._select.selectAll("svg.d3plus-rings").data([0]);
+
+    this._container = this._container.enter().append("svg")
+        .attr("class", "d3plus-rings")
+        .attr("opacity", 0)
+        .attr("width", width)
+        .attr("height", height)
+        .attr("x", this._margin.left)
+        .attr("y", this._margin.top)
+        .style("background-color", "transparent")
+      .merge(this._container);
+
+    this._container.transition(this._transition)
+      .attr("opacity", 1)
+      .attr("width", width)
+      .attr("height", height)
+      .attr("x", this._margin.left)
+      .attr("y", this._margin.top);
+    
+    this._shapes.push(new shapes.Path()
+      .config(configPrep.bind(this)(this._shapeConfig, "edge", "Path"))
+      .id(d => `${d.source.id}_${d.target.id}`)
+      .d(d => d.spline ? `M${d.sourceX},${d.sourceY}C${d.sourceBisectX},${d.sourceBisectY} ${d.targetBisectX},${d.targetBisectY} ${d.targetX},${d.targetY}` : `M${d.source.x},${d.source.y} ${d.target.x},${d.target.y}`)
+      .data(edges)
+      .select(elem("g.d3plus-rings-links", {parent: this._container, transition, enter: {transform}, update: {transform}}).node())
```

Comment:
Unless you specifically need `this._container`, just use `this._select` here and you can ditch lines 416-433.